File: app/src/entourage/java/social/entourage/android/map/entourage/create/wizard/CreateActionWizardPage2Fragment.java
package social.entourage.android.map.entourage.create.wizard;


import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import butterknife.ButterKnife;
import butterknife.OnClick;
import social.entourage.android.Constants;
import social.entourage.android.DrawerActivity;
import social.entourage.android.EntourageActivity;
import social.entourage.android.R;
import social.entourage.android.base.EntourageDialogFragment;
import social.entourage.android.base.EntourageLinkMovementMethod;
import social.entourage.android.tools.Utils;

/**
 * Create Action Wizard Page 2 {@link EntourageDialogFragment} subclass.
 */
public class CreateActionWizardPage2Fragment extends EntourageDialogFragment {

    // ----------------------------------
    // Attributes
    // ----------------------------------

    public static final String TAG = CreateActionWizardPage2Fragment.class.getSimpleName();

    private static final int STEP = 2;

    private CreateActionWizardListener listener;

    // ----------------------------------
    // Lifecycle
    // ----------------------------------


    public CreateActionWizardPage2Fragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_create_action_wizard_page2, container, false);
        ButterKnife.bind(this, view);

        return view;
    }

    @Override
    public void onViewCreated(@NonNull final View view, @Nullable final Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        if (getActivity() != null && getView() != null) {
            TextView disclaimerTextView = getView().findViewById(R.id.create_action_wizard_disclaimer);
            if (disclaimerTextView != null) {
                DrawerActivity activity = (DrawerActivity)getActivity();
                String text = getString(R.string.create_action_wizard_disclaimer, activity.getLink(Constants.CHARTE_LINK_ID));
                disclaimerTextView.setText(Utils.fromHtml(text));
                disclaimerTextView.setMovementMethod(EntourageLinkMovementMethod.getInstance());
            }
        }
    }

    public void setListener(final CreateActionWizardListener listener) {
        this.listener = listener;
    }

    // ----------------------------------
    // Buttons handling
    // ----------------------------------

    @OnClick(R.id.title_close_button)
    protected void onCloseClicked() {
        if (listener != null) listener.createActionWizardPreviousStep(STEP);
        dismiss();
    }

    @OnClick(R.id.create_action_wizard_p2_option1_button)
    protected void onOption1Clicked() {
        if (listener != null) {
            listener.createActionWizardNextStep(STEP, 1);
        }
    }

    @OnClick(R.id.create_action_wizard_p2_option2_button)
    protected void onOption2Clicked() {
        if (listener != null) {
            listener.createActionWizardNextStep(STEP, 2);
        }
    }

}

File: app/src/entourage/java/social/entourage/android/entourage/create/wizard/CreateActionWizardPage2Fragment.kt
package social.entourage.android.entourage.create.wizard

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.entourage.fragment_create_action_wizard_page2.*
import kotlinx.android.synthetic.main.layout_view_title.*
import social.entourage.android.Constants
import social.entourage.android.MainActivity
import social.entourage.android.R
import social.entourage.android.base.EntourageDialogFragment
import social.entourage.android.base.EntourageLinkMovementMethod
import social.entourage.android.tools.Utils

/**
 * Create Action Wizard Page 2 [EntourageDialogFragment] subclass.
 */
class CreateActionWizardPage2Fragment : EntourageDialogFragment() {
    private var listener: CreateActionWizardListener? = null

    // ----------------------------------
    // Lifecycle
    // ----------------------------------
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        super.onCreateView(inflater, container, savedInstanceState)
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_create_action_wizard_page2, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        if (activity != null && getView() != null) {
            val mainActivity = activity as MainActivity
            val text = getString(R.string.create_action_wizard_disclaimer, mainActivity.getLink(Constants.CHARTE_LINK_ID))
            create_action_wizard_disclaimer?.text = Utils.fromHtml(text)
            create_action_wizard_disclaimer?.movementMethod = EntourageLinkMovementMethod.getInstance()
        }
        title_close_button?.setOnClickListener { onCloseClicked()        }
        create_action_wizard_p2_option1_button?.setOnClickListener {onOption1Clicked()}
        create_action_wizard_p2_option2_button?.setOnClickListener {onOption2Clicked()}
    }

    fun setListener(listener: CreateActionWizardListener?) {
        this.listener = listener
    }

    // ----------------------------------
    // Buttons handling
    // ----------------------------------
    fun onCloseClicked() {
        listener?.createActionWizardPreviousStep(STEP)
        dismiss()
    }

    private fun onOption1Clicked() {
        listener?.createActionWizardNextStep(STEP, 1)
    }

    private fun onOption2Clicked() {
        listener?.createActionWizardNextStep(STEP, 2)
    }

    companion object {
        // ----------------------------------
        // Attributes
        // ----------------------------------
        val TAG = CreateActionWizardPage2Fragment::class.java.simpleName
        private const val STEP = 2
    }
}